myhoard/myhoard.py

```diff
@@ -182,6 +172,21 @@ def _restart_systemd(self, *, mysqld_options, service):
             raise Exception(f"Restarting {service!r} failed. Code {e.returncode}") from e
         self.log.info("Restarting %r completed successfully", service)
 
+    @staticmethod
+    def _update_mysqld_opts(env_file_path: str, value: str):
+        key = "MYSQLD_OPTS"
+        try:
+            with open(env_file_path, "r") as f:
+                contents = [
+                    line.rstrip("\n") for line in f.readlines() if line.strip() and not line.startswith(key)
+                ]
+        except FileNotFoundError:
+            contents = []
+        if value is not None:
+            contents.append(f"{key}={value}")
+        with atomic_create_file(env_file_path, binary=True, inherit_owner_from_parent=True) as f:
```

Comment:
Creating a file from scratch would break selinux policies on the file. You either need to call restorecon on it, or better, you only modify the existing file.

Reply:
The `atomic_create_file` function has a bad name. The file should already exist.